lib/rpmsg/rpmsg_virtio.c

```diff
@@ -387,8 +400,16 @@ static void *rpmsg_virtio_get_tx_payload_buffer(struct rpmsg_device *rdev,
 		metal_mutex_release(&rdev->lock);
 		if (rp_hdr || !tick_count)
 			break;
-		metal_sleep_usec(RPMSG_TICKS_PER_INTERVAL);
-		tick_count--;
+
+		/*
+		 * Try to use wait loop implemented in the virtio dispatcher and
+		 * use metal_sleep_usec() method by default.
+		 */
+		status = rpmsg_virtio_notify_wait(rvdev, rvdev->rvq);
+		if (status != RPMSG_SUCCESS) {
```

Comment:
if status is different from RPMSG_EOPNOTSUPP we should leave the loop 

```
if (status == RPMSG_EOPNOTSUPP) {
		metal_sleep_usec(RPMSG_TICKS_PER_INTERVAL);
		tick_count--;
} else if (status == RPMSG_SUCCESS) {
		break;
}
```

Reply:
Ok, done